runtime-light/stdlib/array/array-functions.h

```diff
@@ -5,16 +5,218 @@
 #pragma once
 
 #include <concepts>
+#include <cstdint>
 #include <functional>
 #include <utility>
 
 #include "runtime-common/core/runtime-core.h"
+#include "runtime-common/stdlib/array/array-functions.h"
 #include "runtime-light/coroutine/task.h"
 #include "runtime-light/stdlib/math/random-functions.h"
 #include "runtime-light/utils/concepts.h"
 
+namespace dl {
+
+template<typename T, typename Comparator>
+requires(std::invocable<Comparator, T, T>) task_t<void> sort(T *begin_init, T *end_init, Comparator compare) noexcept {
+  auto compare_call = [compare]<typename U>(U lhs, U rhs) -> task_t<int64_t> {
```

Comment:
Lambda coroutine should not have captured values as they're not saved in a coroutine frame

Reply:
fix